### PR TITLE
Add files to create .deb packages

### PR DESCRIPTION
**Pull request by vatsa287**
Signed-off-by: Shree Vatsa N <vatsa@kadalu.tech>

### CODE REVIEW

**Comment by vatsa287** [comment]
Have to remove some files & edit out default values.

File: debian/kadalu-storage-client.install [hunk 1]
```diff
@@ -0,0 +1,2 @@
+sbin/mount.glusterfs
```
Comment (aravindavk) [inline debian/kadalu-storage-client.install]: mount.glusterfs is not required

File: debian/kadalu-storage-client.manpages [hunk 2]
```diff
@@ -0,0 +1,2 @@
+doc/mount.glusterfs.8
```
Comment (aravindavk) [inline debian/kadalu-storage-client.manpages]: not required

File: debian/kadalu-storage-client.mounting-glusterfs.upstart [hunk 3]
```diff
@@ -0,0 +1,9 @@
+author "Louis Zuckerman <me@louiszuckerman.com>"
```
Comment (aravindavk) [inline debian/kadalu-storage-client.mounting-glusterfs.upstart]: Remove upstart related file.

File: debian/kadalu-storage-common.install [hunk 4]
```diff
@@ -0,0 +1,12 @@
+etc/logrotate.d/*
+etc/glusterfs/gsyncd.conf
```
Comment (aravindavk) [inline debian/kadalu-storage-common.install]: Remove this for now. Related to Geo-rep feature.

File: debian/kadalu-storage-common.install [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+usr/lib/*
+usr/libexec/glusterfs/*
+usr/sbin/glusterfsd
+usr/sbin/gluster-eventsapi
```
Comment (aravindavk) [inline debian/kadalu-storage-common.install]: Remove this as well for now

File: debian/kadalu-storage-common.install [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+usr/sbin/glustereventsd
```
Comment (aravindavk) [inline debian/kadalu-storage-common.install]: Remove this

File: debian/kadalu-storage-common.install [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+usr/sbin/gluster-georep-sshkey
+usr/sbin/gluster-mountbroker
+usr/share/doc/glusterfs/glusterfs-mode.el usr/share/emacs/site-lisp/
```
Comment (aravindavk) [inline debian/kadalu-storage-common.install]: Remove these files

File: debian/kadalu-storage-server.install [hunk 8]
```diff
@@ -0,0 +1,10 @@
+lib/systemd/system/*
+usr/bin/glusterfind
```
Comment (aravindavk) [inline debian/kadalu-storage-server.install]: Remove this

File: debian/kadalu-storage-server.install [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+usr/libexec/glusterfs/glfsheal
+usr/libexec/glusterfs/scripts/gsync-sync-gfid
```
Comment (aravindavk) [inline debian/kadalu-storage-server.install]: Remove this

File: debian/kadalu-storage-server.install [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+usr/sbin/snap_scheduler
```
Comment (aravindavk) [inline debian/kadalu-storage-server.install]: Remove this

File: debian/kadalu-storage-server.install [hunk 11]
```diff
@@ -0,0 +1,10 @@
...
+usr/sbin/gcron
+usr/sbin/gfind_missing_files
```
Comment (aravindavk) [inline debian/kadalu-storage-server.install]: remove this

File: debian/kadalu-storage-server.manpages [hunk 12]
```diff
@@ -0,0 +1 @@
+doc/gluster.8
```
Comment (aravindavk) [inline debian/kadalu-storage-server.manpages]: Remove this. Related to gluster cli

File: debian/rules [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+	dh_install --list-missing
+	mkdir -p $(CURDIR)/debian/glusterfs-common/usr/lib/glusterfs
+	rm -rf $(CURDIR)/debian/glusterfs-common/usr/libexec/glusterfs/glfsheal
+	rm -rf $(CURDIR)/debian/glusterfs-common/usr/libexec/glusterfs/scripts/gsync-sync-gfid
+
+
+override_dh_python3:
+	dh_python3 usr/libexec/glusterfs/python/syncdaemon/
+	dh_python3 usr/libexec/glusterfs/glusterfind/
+	dh_python3 usr/libexec/glusterfs/gfevents/
```
Comment (aravindavk) [inline debian/rules]: Remove this

File: debian/rules [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+	mv debian/tmp/usr/sbin/snap_scheduler.py debian/tmp/usr/sbin/snap_scheduler
+	install -c -d -m 0755 debian/tmp/run/gluster/metrics
+	dh_install --list-missing
+	mkdir -p $(CURDIR)/debian/glusterfs-common/usr/lib/glusterfs
+	rm -rf $(CURDIR)/debian/glusterfs-common/usr/libexec/glusterfs/glfsheal
+	rm -rf $(CURDIR)/debian/glusterfs-common/usr/libexec/glusterfs/scripts/gsync-sync-gfid
+
+
+override_dh_python3:
+	dh_python3 usr/libexec/glusterfs/python/syncdaemon/
```
Comment (aravindavk) [inline debian/rules]: remove this

File: debian/rules [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+	dh_python3 usr/libexec/glusterfs/glusterfind/
+	dh_python3 usr/libexec/glusterfs/gfevents/
+
+override_dh_installinit:
+	dh_installinit -p kadalu-storage-server --name=glusterta-volume
+	dh_installinit -p kadalu-storage-server --name=glustereventsd
+	dh_installinit -p kadalu-storage-server --name=glusterfssharedstorage
```
Comment (aravindavk) [inline debian/rules]: remove this

File: debian/rules [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+		debian/tmp/lib/systemd/system/gluster-ta-volume.service
+	install -D -p -m 0644 extras/systemd/glustereventsd.service \
+		debian/tmp/lib/systemd/system/glustereventsd.service
+	mv debian/tmp/usr/sbin/gcron.py debian/tmp/usr/sbin/gcron
+	mv debian/tmp/usr/sbin/snap_scheduler.py debian/tmp/usr/sbin/snap_scheduler
+	install -c -d -m 0755 debian/tmp/run/gluster/metrics
+	dh_install --list-missing
+	mkdir -p $(CURDIR)/debian/glusterfs-common/usr/lib/glusterfs
+	rm -rf $(CURDIR)/debian/glusterfs-common/usr/libexec/glusterfs/glfsheal
+	rm -rf $(CURDIR)/debian/glusterfs-common/usr/libexec/glusterfs/scripts/gsync-sync-gfid
```
Comment (aravindavk) [inline debian/rules]: Remove

File: debian/rules [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+		debian/tmp/etc/logrotate.d/glusterfs-georep
+	install -D -p -m 0755 extras/mount-shared-storage.sh \
+		debian/tmp/usr/libexec/glusterfs/mount-shared-storage.sh
+	install -c -d -m 755 debian/tmp/lib/systemd/system
+	install -D -p -m 0644 extras/systemd/gluster-ta-volume.service \
+		debian/tmp/lib/systemd/system/gluster-ta-volume.service
+	install -D -p -m 0644 extras/systemd/glustereventsd.service \
+		debian/tmp/lib/systemd/system/glustereventsd.service
+	mv debian/tmp/usr/sbin/gcron.py debian/tmp/usr/sbin/gcron
+	mv debian/tmp/usr/sbin/snap_scheduler.py debian/tmp/usr/sbin/snap_scheduler
```
Comment (aravindavk) [inline debian/rules]: remove

File: debian/rules [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+	install -D -p -m 0644 extras/glusterfs-georep-logrotate \
+		debian/tmp/etc/logrotate.d/glusterfs-georep
+	install -D -p -m 0755 extras/mount-shared-storage.sh \
+		debian/tmp/usr/libexec/glusterfs/mount-shared-storage.sh
+	install -c -d -m 755 debian/tmp/lib/systemd/system
+	install -D -p -m 0644 extras/systemd/gluster-ta-volume.service \
+		debian/tmp/lib/systemd/system/gluster-ta-volume.service
+	install -D -p -m 0644 extras/systemd/glustereventsd.service \
+		debian/tmp/lib/systemd/system/glustereventsd.service
+	mv debian/tmp/usr/sbin/gcron.py debian/tmp/usr/sbin/gcron
```
Comment (aravindavk) [inline debian/rules]: remove

File: debian/rules [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+	install -D -p -m 0644 extras/glusterfs-logrotate \
+		debian/tmp/etc/logrotate.d/glusterfs
+	install -D -p -m 0644 extras/glusterfs-georep-logrotate \
+		debian/tmp/etc/logrotate.d/glusterfs-georep
+	install -D -p -m 0755 extras/mount-shared-storage.sh \
+		debian/tmp/usr/libexec/glusterfs/mount-shared-storage.sh
+	install -c -d -m 755 debian/tmp/lib/systemd/system
+	install -D -p -m 0644 extras/systemd/gluster-ta-volume.service \
+		debian/tmp/lib/systemd/system/gluster-ta-volume.service
+	install -D -p -m 0644 extras/systemd/glustereventsd.service \
```
Comment (aravindavk) [inline debian/rules]: remove

File: debian/rules [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+	strip --remove-section=.comment --remove-section=.note --strip-unneeded \
+		debian/tmp/usr/lib/*/glusterfs/*/xlator/debug/error-gen.so
+	strip --remove-section=.comment --remove-section=.note --strip-unneeded \
+		debian/tmp/usr/lib/*/glusterfs/*/xlator/debug/io-stats.so
+	find debian/tmp/usr/lib -type f -name \*.la -exec sed 's|^dependency_libs|#dependency_libs|g' -i {} \;
+	install -D -p -m 0644 extras/glusterfs-logrotate \
+		debian/tmp/etc/logrotate.d/glusterfs
+	install -D -p -m 0644 extras/glusterfs-georep-logrotate \
+		debian/tmp/etc/logrotate.d/glusterfs-georep
+	install -D -p -m 0755 extras/mount-shared-storage.sh \
```
Comment (aravindavk) [inline debian/rules]: remove

File: debian/control [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+Pre-Depends: dpkg (>= 1.15.6~)
+Depends: ${misc:Depends},
+ ${shlibs:Depends},
+ python3,
+ fuse,
+ kadalu-storage-common (>= ${binary:Version}),
+Description: clustered file-system (client package)
+ This package provides the FUSE based Kadalu-Storage/GlusterFS tool.
+
+Package: kadalu-storage-server
```
Comment (aravindavk) [inline debian/control:46]: Add conflicts glusterfs-server

**Comment by vatsa287** [comment]
Sent to moana repo.

**Pull request closed without merging**